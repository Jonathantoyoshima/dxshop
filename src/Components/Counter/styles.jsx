import styled from 'styled-components';

export const Root = styled.div`
  position: absolute;
  top: -2px;
  right: -5px;
  background-color: ${(props) => props.theme.color.secondary.light};
  color: ${(props) => props.theme.color.basic.white};
  font-size: 0.8rem;
  border-radius: 30px;
  min-width: 18px;
  height: 18px;
  display: ${(props) => (props.hidden ? 'flex' : 'none')};
  align-items: center;
  justify-content: center;
  font-weight: bold;
`;
