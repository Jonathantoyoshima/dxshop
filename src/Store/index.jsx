import React from 'react';

const categories = [
  { id: 0, name: 'Bebidas' },
  { id: 1, name: 'Doces' },
  { id: 2, name: 'Salgados' },
];

const data = [
  { id: 0, idCategory: 0, name: 'Coca-cola lata 350ml', price: 3.5 },
  { id: 1, idCategory: 0, name: 'Fanta lata 350ml', price: 3.5 },
  { id: 2, idCategory: 1, name: 'Bolinho Ana maria', price: 1.5 },
  { id: 3, idCategory: 2, name: 'Ruffes Pequeno 350g', price: 5.0 },
];

const cart = [];

const info = { data: data, categories: categories, cart: cart, chooseCategory: -1 };

export const StoreContext = React.createContext(info);

export const Store = ({ children }) => {
  const [context, setContext] = React.useState(info);

  return <StoreContext.Provider value={[context, setContext]}>{children}</StoreContext.Provider>;
};
