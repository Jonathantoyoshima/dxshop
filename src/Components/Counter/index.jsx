import React from 'react';
import { Root } from './styles';
import { StoreContext } from 'Store';
const defaultProps = {
  theme: {
    color: {
      secondary: { light: '#0ff' },
      basic: { white: '#fff' },
    },
  },
};

const Counter = () => {
  const [store] = React.useContext(StoreContext);
  return <Root hidden={store.cart.length > 0}>{store.cart.length}</Root>;
};

Root.defaultProps = defaultProps;

export default Counter;
