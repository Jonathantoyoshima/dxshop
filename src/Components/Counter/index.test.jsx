import React from 'react';
import Counter from './index';
import { shallow } from 'enzyme';
import { Store } from 'Store';

test('Render Build Counter Snapshot', () => {
  const component = shallow(
    <Store>
      <Counter />
    </Store>
  );
  expect(component).toMatchSnapshot();
});
