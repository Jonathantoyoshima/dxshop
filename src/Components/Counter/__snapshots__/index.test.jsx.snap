// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Build Counter Snapshot 1`] = `ShallowWrapper {}`;
